.github/workflows/lint.yaml

```diff
@@ -0,0 +1,43 @@
+name: "lint"
+
+on:
+  pull_request:
+    types: ["opened", "reopened"]
+  push:
+  schedule:
+    # Run at midnight UTC every day with 15 minutes delay added to avoid high load periods
+    - cron: "15 0 * * *"
+  workflow_dispatch:
+
+permissions:
+  # So the workflow can cancel in-progress jobs
+  actions: "write"
+  # So `eslint-annotate-action` can create / update status checks
+  checks: "write"
+  # So `eslint-annotate-action` can get pull request files
+  contents: "read"
+
+concurrency:
+  group: "${{github.workflow}}-${{github.ref}}"
+  # Cancel in-progress jobs for efficiency
+  cancel-in-progress: true
+
+jobs:
+  lint-check:
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: "actions/checkout@v4"
+        with:
+          submodules: "recursive"
+      - uses: "actions/setup-node@v4"
+        with:
+          node-version: 22
+      - run: "npm --prefix new-log-viewer/ clean-install"
+      - run: "npm --prefix new-log-viewer/ run lint:ci"
+        continue-on-error: true
+      - uses: "ataylorme/eslint-annotate-action@v3"
+        with:
+          fail-on-error: true
+          fail-on-warning: true
+          only-pr-files: true
+          report-json: "./new-log-viewer/eslint_report.json"
```

Comment:
Since we're setting the path explicitly, can we use `eslint-report.json` (hyphen instead of underscore)?